File: src/main/java/eu/siacs/conversations/ui/AboutPreference.java
package eu.siacs.conversations.ui;

import android.content.Context;
import android.content.Intent;
import android.os.Build;
import android.preference.Preference;
import android.util.AttributeSet;

import com.google.common.base.Strings;

import eu.siacs.conversations.BuildConfig;
import eu.siacs.conversations.R;

public class AboutPreference extends Preference {
    public AboutPreference(final Context context, final AttributeSet attrs, final int defStyle) {
        super(context, attrs, defStyle);
        setSummaryAndTitle(context);
    }

    public AboutPreference(final Context context, final AttributeSet attrs) {
        super(context, attrs);
        setSummaryAndTitle(context);
    }

    private void setSummaryAndTitle(final Context context) {
        final String appName = context.getString(R.string.app_name);
        setSummary(String.format("%s %s %s (%s)", appName, BuildConfig.VERSION_NAME, im.conversations.webrtc.BuildConfig.WEBRTC_VERSION, Strings.nullToEmpty(Build.DEVICE)));
        setTitle(context.getString(R.string.title_activity_about_x, appName));
    }

    @Override
    protected void onClick() {
        super.onClick();
        final Intent intent = new Intent(getContext(), AboutActivity.class);
        getContext().startActivity(intent);
    }
}

File: src/main/java/eu/siacs/conversations/ui/AboutPreference.kt
package eu.siacs.conversations.ui

import android.content.Context
import android.content.Intent
import android.util.AttributeSet
import androidx.preference.Preference
import androidx.preference.PreferenceViewHolder
import eu.siacs.conversations.ui.activity.DeveloperOptionsActivity

/**
 * Hidden entry point: long-pressing this row opens Developer Options. Uses AndroidX's
 * [androidx.preference.Preference] (not the legacy `android.preference.Preference`) because
 * the settings screen is hosted by [androidx.preference.PreferenceFragmentCompat], which can
 * only inflate custom preferences from that hierarchy.
 */
class AboutPreference : Preference {
    constructor(context: Context, attrs: AttributeSet?, defStyle: Int) : super(context, attrs, defStyle)
    constructor(context: Context, attrs: AttributeSet?) : super(context, attrs)
    constructor(context: Context) : super(context)

    override fun onBindViewHolder(holder: PreferenceViewHolder) {
        super.onBindViewHolder(holder)
        holder.itemView.setOnLongClickListener {
            it.context.startActivity(Intent(it.context, DeveloperOptionsActivity::class.java))
            true
        }
    }
}
